File: Burrow/Shovel/ResourceRecord.m
//
//  Test.m
//  Burrow
//
//  Created by Jaden Geller on 4/10/16.
//
//

#import "ResourceRecord.h"

NSInteger ResourceRecordGetDataLength(ResourceRecord record) {
    return ns_rr_rdlen(record);
}

const u_char *ResourceRecordGetData(ResourceRecord record) {
    return ns_rr_rdata(record);
}

ns_type ResourceRecordGetType(ResourceRecord record) {
    return ns_rr_type(record);
}

File: Burrow/Shovel/ResourceRecord.swift
//
//  ResourceRecord.swift
//  Burrow
//
//  Created by Jaden Geller on 4/10/16.
//
//

extension ResourceRecord {
    var dataBuffer: UnsafeBufferPointer<UInt8> {
        return UnsafeBufferPointer(
            start: ResourceRecordGetData(self),
            count: ResourceRecordGetDataLength(self)
        )
    }
}

extension RecordType {
    static var txt: RecordType {
        return ns_t_txt
    }
}

extension RecordClass {
    static var internet: RecordClass {
        return ns_c_in
    }
}